dashboard/app/controllers/pd/professional_learning_landing_controller.rb

```diff
@@ -4,11 +4,18 @@ class Pd::ProfessionalLearningLandingController < ApplicationController
   def index
     # Get the courses that this user teaches
     workshops = Pd::Workshop.enrolled_in_by(current_user)
+    ended_workshops = workshops.in_state(Pd::Workshop::STATE_ENDED)
+
     courses_teaching = workshops.pluck(:course).uniq
+    courses_completed = ended_workshops.pluck(:course).uniq
+
+    last_enrollment =  Pd::Enrollment.joins(:workshop).where(email: current_user.email).where('pd_enrollments.completed_survey_id IS NULL').order('pd_workshops.ended_at DESC').first
 
     # Link to the certificate
     @landing_page_data = {
-      courses_teaching: courses_teaching
+      courses_teaching: courses_teaching,
+      courses_completed: courses_completed,
+      last_workshop_survey_url: "#{CDO.code_org_url}/pd-workshop-survey/#{last_enrollment.code}"
```

Comment:
In order for this survey_url to be valid, the workshop needs to be ended. The `last_enrollment` query above needs more constraints, and a more descriptive name.

What if there are no enrollments? Line 18 will fail accessing `.code` on nil.

Also this line should be 
```ruby
  last_workshop_survey_url: CDO.code_org_url("/pd-workshop-survey/#{last_enrollment.code})"
```